File: src/my_project/interviews/top_150_questions_round_22/ex_53_evaluate_reverse_polish_notation.py
from typing import List, Union, Collection, Mapping, Optional
from abc import ABC, abstractmethod

class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []
        operators = ['+', '-', '*', '/']
        
        for token in tokens:
            if token in operators:
                # Pop two operands (note: order matters for - and /)
                b = stack.pop()
                a = stack.pop()
                
                # Apply operation
                if token == '+':
                    result = a + b
                elif token == '-':
                    result = a - b
                elif token == '*':
                    result = a * b
                else:  # token == '/'
                    # Truncate toward zero
                    result = int(a / b)
                
                stack.append(result)
            else:
                # It's a number, push to stack
                stack.append(int(token))
        
        # Final result is the only element left in stack
        return stack[0]

File: src/my_project/interviews_typescript/top_150_questions_round_1/ex_53_evaluate_reverse_polish_notation.ts
function evalRPN(tokens: string[]): number {
    const stack: number[] = [];
    const operators: string[] = ['+', '-', '*', '/'];
    
    for (const token of tokens) {
        if (operators.includes(token)) {
            // Pop two operands (note: order matters for - and /)
            const b = stack.pop()!;
            const a = stack.pop()!;
            
            // Apply operation
            let result: number;
            if (token === '+') {
                result = a + b;
            } else if (token === '-') {
                result = a - b;
            } else if (token === '*') {
                result = a * b;
            } else { // token === '/'
                // Truncate toward zero
                result = Math.trunc(a / b);
            }
            
            stack.push(result);
        } else {
            // It's a number, push to stack
            stack.push(parseInt(token));
        }
    }
    
    // Final result is the only element left in stack
    return stack[0];
}